tests/PathMappingRepositoryTest.php

```diff
@@ -131,6 +131,38 @@ public function testResolveFilesystemResource()
         $this->assertEquals(Path::normalize(__DIR__.'/Fixtures/dir5/sub'), $resource->getFilesystemPath());
     }
 
+    public function testMergeMixedParent()
```

Comment:
I'm not sure but I think you are still missing a test case here. If you're resolving `/sub1/sub2/sub3`, there are a few possibilities:

- `/sub1/sub2/sub3` exists (tested in other tests)
- `/sub1/sub2/sub3` doesn't exists but `/sub1/sub2` exists => you check the resource `/dir2/sub3`
- neither `/sub1/sub2/sub3` nor `/sub1/sub2` exist but `/sub1` exists => you check the resource `/dir1/sub2/sub3`

I think you didn't check all these possibilities. If you want, I can write the test I have in mind in a gist. Perhaps am I misunderstanding your code though, if you think I'm wrong please tell me :) !